core/domain/stats_services.py

```diff
@@ -47,36 +48,65 @@ def get_exps_unresolved_answers_count_for_default_rule(exp_ids):
+              {'count': 4, value: 'answer_1'},
+              {'count': 2, value: 'answer_2'},
+              {'count': 1, value: 'answer_3'}
+            ]
+          },
+          'exp_id_2': {
+            'count': (number of unresolved answers for this exploration),
+            'values': [
+              {'count': 8, value: 'answer_4'},
+              {'count': 3, value: 'answer_5'},
+              {'count': 1, value: 'answer_6'}
+            ]
+          }
+        }
     """
-    explorations = exp_services.get_multiple_explorations_by_id(exp_ids)
-
-    # The variable `exploration_states_tuples` is a list of all
-    # (exp_id, state_name) tuples for the given exp_ids.
-    # E.g. - [
-    #   ('eid1', 'Introduction'),
-    #   ('eid1', 'End'),
-    #   ('eid2', 'Introduction'),
-    #   ('eid3', 'Introduction')
-    # ]
-    # when exp_ids = ['eid1', 'eid2', 'eid3'].
-    explorations_states_tuples = [
-        (exp_domain_object.id, state_key)
-        for exp_domain_object in explorations.values()
-        for state_key in exp_domain_object.states
-    ]
+    def _get_explorations_state_tuples_by_ids(exp_ids):
+        """Returns a list of all (exp_id, state_name) tuples for the given
+        exp_ids.
+        E.g. - [
+          ('eid1', 'Introduction'),
+          ('eid1', 'End'),
+          ('eid2', 'Introduction'),
+          ('eid3', 'Introduction')
+        ]
+        when exp_ids = ['eid1', 'eid2', 'eid3']."""
+        explorations = exp_services.get_multiple_explorations_by_id(exp_ids)
+        return [
+            (exp_domain_object.id, state_key)
+            for exp_domain_object in explorations.values()
+            for state_key in exp_domain_object.states
+        ]
+
+    explorations_states_tuples = _get_explorations_state_tuples_by_ids(exp_ids)
     exploration_states_answers_list = get_top_state_rule_answers_multi(
         explorations_states_tuples, [exp_domain.DEFAULT_RULESPEC_STR])
     exps_answers_mapping = {}
 
     for ind, statewise_answers in enumerate(exploration_states_answers_list):
+        exp_id = explorations_states_tuples[ind][0]
+        if exp_id not in exps_answers_mapping:
+            exps_answers_mapping[exp_id] = {
+                'count': 0,
+                'values': []
+            }
+            exps_answers_mapping[exp_id]['values'].extend(statewise_answers)
         for answer in statewise_answers:
-            exp_id = explorations_states_tuples[ind][0]
-            if exp_id not in exps_answers_mapping:
-                exps_answers_mapping[exp_id] = 0
-            exps_answers_mapping[exp_id] += answer['count']
+            exps_answers_mapping[exp_id]['count'] += answer['count']
+
+    for exp_id in exps_answers_mapping:
+        exps_answers_mapping[exp_id]['values'] = (
+            sorted(
+                exps_answers_mapping[exp_id]['values'],
+                key=lambda a: a['count'],
+                reverse=True)[:feconf.TOP_UNRESOLVED_ANSWERS_COUNT_DASHBOARD])
```

Comment:
Agreed about (1).
For (2), any special reasons for keeping this constant in dashboard.py instead of feconf.py ?